src/response/app_keys_list.rs

```diff
@@ -0,0 +1,101 @@
+use std::prelude::v1::*;
+
+use std::fmt;
+
+use serde::{
+    de::{Deserializer, MapAccess, Visitor},
+    Deserialize,
+};
+
+use crate::{
+    application::{AppId, AppKey},
+    credentials::ServiceId,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AppKeysList {
+    service: Option<ServiceId>,
```

Comment:
Why are `service` and `app` optional?

Reply:
Because it depended on what shape this would have taken in Apisonator, and whether we'd end up requiring them to be always present or not. Additionally using `Option` has the benefit that if we ever decided they are superfluous and removed them, this would allow the code to still parse the keys without bailing out, which is really the main raison-d'etre of this extension.

Reply:
Fair enough. These are not optional right now in Apisonator, so it might be a good idea to add a comment to clarify that. I'll leave it to you.